src/OpenTK.Compute/OpenCL/CLGL.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Runtime.InteropServices;
+using OpenTK.Compute.Native;
+
+namespace OpenTK.Compute.OpenCL
+{
+	public class CLGL : CLBase
+	{
+		public enum ObjectType : uint
+		{
+			ObjectBuffer = 0x2000,
+			ObjectTexture2D = 0x2001,
+			ObjectTexture3D = 0x2002,
+			ObjectRenderBuffer = 0x2003,
+
+			/// <summary>
+			/// Introduced in OpenCL 1.2
+			/// </summary>
+			ObjectTexture2DArray = 0x200E,
+
+			/// <summary>
+			/// Introduced in OpenCL 1.2
+			/// </summary>
+			ObjectTexture1D = 0x200F,
+
+			/// <summary>
+			/// Introduced in OpenCL 1.2
+			/// </summary>
+			ObjectTexture1DArray = 0x2010,
+
+			/// <summary>
+			/// Introduced in OpenCL 1.2
+			/// </summary>
+			ObjectTextureBuffer = 0x2011,
+		}
+
+		public enum TextureInfo : uint
+		{
+			TextureTarget = 0x2004,
+			MipmapLevel = 0x2005,
+
+			/// <summary>
+			/// Introduced in OpenCL 1.2
+			/// </summary>
+			NumSamples = 0x2012,
+		}
+
+		static CLGL()
+		{
+			CLBase.RegisterOpenCLResolver();
+		}
+
+		private const string LibName = "libOpenCL";
+		private const CallingConvention CallingConvention = System.Runtime.InteropServices.CallingConvention.Cdecl;
+
+		/// <summary>
+		/// Introduced in OpenCL 1.0
+		/// </summary>
+		[DllImport(LibName, CallingConvention = CallingConvention, EntryPoint = "clCreateFromGLBuffer")]
+		public static extern CLBuffer CreateFromGLBuffer(
+			[In] CLContext context,
+			[In] MemoryFlags flags,
+			[In] int glBuffer,
+			[Out] out CLResultCode error
+		);
+
+		/// <summary>
+		/// Introduced in Opencl 1.2
+		/// </summary>
+		[DllImport(LibName, CallingConvention = CallingConvention, EntryPoint = "clCreateFromGLTexture")]
+		public static extern CLBuffer CreateFromGLTexture(
+			[In] CLContext context,
+			[In] MemoryFlags flags,
+			//TODO FIX
```

Comment:
What is there to fix here?

Reply:
For documentation purposes I will write down what the FIXME is about here.
This argument is supposed to be an enum that is defined in opengl, but currently the OpenCL bindings don't have a dependency on OpenGL so we don't have access to that enum type here.

One solution we could consider in the future is splitting the CLxGL interop into it's own package which takes a dependency on both the CL and GL bindings. But for now I think this is fine.